### PR TITLE
Final Working Version (Yay!)

### PR DESCRIPTION
**Pull request by HadikHuszar**
Please let me know what you think needs improvement!

### CODE REVIEW

**Comment by HadikHuszar** [comment]
The file to check out is:  1.  App.js;

File: Knowledge_Check_Wk08/client/src/App.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+	useEffect(() => {
+	    if (resetGame){
+		setShowScore(false);
+		setScore(0);
+		setCurrentQuestion(0);
+		setResetGame(false);
+		}
+	}, [resetGame])
+
```
Comment (xiaozhong21) [inline Knowledge_Check_Wk08/client/src/App.js:41]: Very nice use of useEffect to control and flow of the game!!

File: Knowledge_Check_Wk08/client/src/App.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+//// something to do while there is no data
+
+	if (questions.length === 0) {
+		return (
+			<div className='app'>
+				"loading ..."
+			</div>
+		)
+	}
+
```
Comment (xiaozhong21) [inline Knowledge_Check_Wk08/client/src/App.js:68]: Super thoughtful here - love how you are thinking about the potential errors or network problems!!

File: Knowledge_Check_Wk08/client/src/App.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+				<>
+					<div className='question-section'>
+						<div className='question-count'>
+							<span>Question {currentQuestion + 1}</span>/{questions.length}
+						</div>
+						<div className='question-text'>{questions[currentQuestion].title}</div>
+					</div>
+					<div className='answer-section'>
+						{questions[currentQuestion].choices.map((choice, index) => 
+							<Button key={`button_${index}`} buttonAnswer={choice} handleAnswerButtonClick={handleAnswerButtonClick}/>
```
Comment (xiaozhong21) [inline Knowledge_Check_Wk08/client/src/App.js:91]: Great use of array.map() method here - making the code so much more concise and efficient!!

**Review by xiaozhong21 — APPROVED** [review]
I love how smooth your app is!! And very efficient and clean code. Way to go Elizabeth!!

**Pull request merged**